ci_framework/roles/tempest/tasks/main.yml

```
@@ -0,0 +1,44 @@
+---
+# Copyright Red Hat, Inc.
+# All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License. You may obtain
+# a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations
+# under the License.
+
+- name: Ensure podman is installed
+  become: true
+  ansible.builtin.package:
+    name: podman
+    state: present
+
+- name: Create tempest directories
+  ansible.builtin.file:
+    path: "{{ cifmw_tempest_artifacts_basedir }}"
+    state: directory
+
+- name: Setup tempest tests
+  ansible.builtin.include_tasks: tempest-tests.yml
+
+- name: Configure tempest
+  ansible.builtin.include_tasks: configure-tempest.yml
+  when: not cifmw_tempest_dry_run | bool
+
+- name: Set proper permission for tempest directory
+  ansible.builtin.shell:
+    cmd: "podman unshare chown 42480:42480 -R {{ cifmw_tempest_artifacts_basedir }}"
+  when: not cifmw_tempest_dry_run | bool
+
+# > {{ cifmw_tempest_artifacts_basedir }}/tempest_output.log"
+- name: Run tempest
+  ansible.builtin.command:
```

```suggestion
  containers.podman.podman_container:
       name: tempest
       image: "{{ cifmw_tempest_image }}:{{ cifmw_tempest_image_tag }}"
       state: started
       volume: 
          - "{{ cifmw_tempest_artifacts_basedir }}/:/var/lib/tempest/external_files:Z"
```
Can we use [ansible containers module here](https://docs.ansible.com/ansible/latest/collections/containers/podman/podman_container_module.html#containers-podman-podman-container-module-manage-podman-containers )